io-engine/src/bdev/nexus/nexus_persistence.rs

```diff
@@ -103,6 +103,15 @@ impl<'n> Nexus<'n> {
                     };
                     nexus_info.children.push(child_info);
                 });
+                // We started with this child because it was healthy in etcd, or isn't there at all.
+                // Being unhealthy here means it is undergoing a fault/retire before nexus is open.
+                if nexus_info.children.len() == 1 && !nexus_info.children[0].healthy {
```

Comment:
How do we get here?
We create the nexus, and open the children (or child in this case) and when we're saving the nexus info the child gets faulted at the same time?

Reply:
yeah, the child is undergoing retire path while nexus creation isn't complete. We see the persist call from retire path emits this warning since the create path persisted replica as unhealthy.
>[2024-11-13T18:06:02.134805509+00:00  WARN io_engine::bdev::nexus::nexus_bdev_children:nexus_bdev_children.rs:1055] Nexus 'ba4de70e-0ecb-4488-bb2a-40abb924a634' [open]: no healthy replicas persent in persistent store when retiring replica 'nvmf://x.x.x.x:8420/nqn.2019-05.io.openebs:ef00f725-b47d-4720-aa69-184cf3ab0114?uuid=ef00f725-b47d-4720-aa69-184cf3ab0114':
                            not persisting the replica state